### ISSUE TITLE
Create a user in the database before the verification code!!

### ISSUE THREAD

**Issue by mhghr** [lucia-auth/example-nextjs-email-password-2fa#3]
Hello,
I've noticed an issue in the user registration process. It seems that the user record is being created in the database before the user verifies the email code sent to them. Shouldn't the user account be created only after successful verification of the email code?

This can potentially lead to inconsistencies or unauthorized accounts being stored in the database.

Looking forward to your clarification or a fix for this.

**Comment by ambergristle** [comment]
hey @mhghr! i'd recommend taking questions like this to the Lucia Auth discord: https://discord.com/invite/PwrK3kpVR3

could you expand a bit on what you mean by
> inconsistencies or unauthorized accounts being stored in the database

Lucia Auth uses sessions/cookies to manage workflows like registration. if a user doesn't have a valid `User`, `AuthSession`, and `EmailVerificationSession` record (which are all linked), they can't proceed with registration/verification.
- at this point, they're already partially authorized; but flags on the `User` record require the flow is completed

it's true that you'd be responsible for cleaning up abandoned registration flows/user records, but that seems out of scope for the example. it also doesn't seem like a big deal from a security perspective: users can pick up the process where they left off, but only with the email/password they started with.
